include/deal.II/base/polynomials_base.h

```diff
@@ -0,0 +1,169 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2005 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// NOTE: The required calls for FE_PolyTensor seem to be
+// poly_space.compute(), poly_space.n(), and poly_space.degree()
+// poly_space.n() will be filled by the function compute_n_pols(k).
+
+#ifndef dealii_polynomials_base_h
+#define dealii_polynomials_base_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/polynomial_space.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/tensor_product_polynomials.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class provides a framework for the finite element polynomial
+ * classes for use with finite element classes that are derived from
+ * @p FE_Poly and @p FE_PolyTensor. An object of this type is stored
+ * as a member variable in the finite element class.
+ *
+ * <h3>Deriving classes</h3>
+ *
+ * Any derived class must provide a the most basic properties for shape
```

Comment:
It might be useful to have a section about writing derived classes when others try to implement their own finite element classes. Is there a way I can make this section more meaningful?

Reply:
It's useful in general, but just saying that derived classes have to implement abstract virtual functions is pretty obvious :-) If you want to, provide an overview of what kinds of operations (and in which functions) derived classes have to implement.

Reply:
You might also point out which classes use this class, and what functions they use.

Reply:
Okay. Let me know how this update looks. I tried to improve some of the wording too.